mlserver/batch_processing.py

```diff
@@ -81,87 +82,155 @@ def get_headers(request_id: str) -> Dict[str, str]:
     return headers
 
 
-def json_to_triton(
-    request_json: bytes, binary_data: bool
-) -> Tuple[str, List[httpclient.InferInput], List[httpclient.InferRequestedOutput]]:
-    inputs = []
-    inference_request = InferenceRequest.parse_obj(orjson.loads(request_json))
-    for request_input in inference_request.inputs or []:
-        new_input = httpclient.InferInput(
-            request_input.name, request_input.shape, request_input.datatype
-        )
-        new_input.set_data_from_numpy(
-            NumpyCodec.decode_input(request_input),
-            binary_data=binary_data,
-        )
-        inputs.append(new_input)
+@dataclass
+class TritonRequest:
+    id: str
+    inputs: List[httpclient.InferInput]
+    outputs: List[httpclient.InferRequestedOutput] = None
+
+    @classmethod
+    def from_inference_request(
+        cls, inference_request: InferenceRequest, binary_data: bool
+    ) -> Tuple[str, List[httpclient.InferInput], List[httpclient.InferRequestedOutput]]:
+        inputs = []
+
+        for request_input in inference_request.inputs or []:
+            new_input = httpclient.InferInput(
+                request_input.name, request_input.shape, request_input.datatype
+            )
+            new_input.set_data_from_numpy(
+                NumpyCodec.decode_input(request_input),
+                binary_data=binary_data,
+            )
+            inputs.append(new_input)
 
-    outputs = []
-    for request_output in inference_request.outputs or []:
-        new_output = httpclient.InferRequestedOutput(
-            request_output.name, binary_data=binary_data
-        )
-        outputs.append(new_output)
+        outputs = []
+        for request_output in inference_request.outputs or []:
+            new_output = httpclient.InferRequestedOutput(
+                request_output.name, binary_data=binary_data
```

Comment:
Out of curiosity, do you know what the `binary_data` kwarg does here?

Reply:
Yes, I do! It instructs Triton to return responses as `binary_data` or not. However, [this will only work if we specify the actual parameters we want back](https://github.com/triton-inference-server/client/blob/84756de4f8245d4d333bfb91d2879a922c0b64e7/src/python/library/tritonclient/http/__init__.py#L100-L107), otherwise `tritonclient` will assume we always want binary data back (no harm as it will know how deal with them).

Reply:
Interesting! That's good to know.